packages/roosterjs-editor-core/lib/editor/Editor.ts

```diff
@@ -170,7 +174,52 @@ export default class Editor {
      * @returns true if node is inserted. Otherwise false
      */
     public insertNode(node: Node, option?: InsertOption): boolean {
-        return node ? this.core.api.insertNode(this.core, node, option) : false;
+        let darkModeNormalize = null;
+        if (this.isDarkMode()) {
+            darkModeNormalize = this.convertContentToDarkMode(node);
+        }
+
+        const result = node ? this.core.api.insertNode(this.core, node, option) : false;
+
+        if (result && darkModeNormalize) {
+            darkModeNormalize();
+        }
+        return result;
+    }
+
+    private isElement(node: Node | HTMLElement): node is HTMLElement {
+        return (<HTMLElement>node).getElementsByTagName !== undefined && (<HTMLElement>node).style !== undefined;
+    }
+
+    private isParentNode(node: Node | ParentNode): node is ParentNode {
+        return (<ParentNode>node).querySelectorAll !== undefined;
+    }
+
+    private convertContentToDarkMode(node: Node): () => void {
+        let childElements: HTMLElement[];
+        // For dark mode, we must ammass a list of all the decendents of a node.
+        // To do this, we have to take mulitple tactics, as a Node could be literally anything.
+        // If a node is an element, getElementsByTagName will return a list of HTML elements.
+        // If it's not an element, it could be a DocumentFragment. querySelectorAll has broad support and achieves the same result.
+        // If it's neither of these things, just give up.
+        if (this.isElement(node)) {
+            childElements = Array.prototype.slice.call(node.getElementsByTagName('*'));
+            childElements.unshift(node);
+        } else if (this.isParentNode(node)) {
+            childElements = Array.prototype.slice.call(node.querySelectorAll('*'));
+        }
+
+        return () => {
+            childElements.forEach((element) => {

```

Comment:
This will error on a node that has no children (e.g. a Text node) because childElements will not be initialized.

Reply:
I'll initialize a default empty array so this won't occur.